pkg/core/blockchain.go

```diff
@@ -477,6 +477,25 @@ func (bc *Blockchain) init() error {
 	return bc.updateExtensibleWhitelist(bHeight)
 }
 
+func (bc *Blockchain) removeOldStorageItems() {
+	stage, err := bc.dao.Store.Get(storage.SYSStateJumpStage.Bytes())
```

Comment:
There's a *very tiny* chance that this goroutine will start after `jumpstage` flag is removed from the storage inside `(*Blockchain).jumpToState()` method. If so, then old storage items will never be removed. So maybe we'd better move `jumpstage` check out of this method to `(bc *Blockchain) init()`? Just in case.

Reply:
In your solution there is still a chance that we start iterating but won't finish before node removes the key and crashes, right?
I think we can just seek and remove: the flag is missing if sync is finished or is disabled. In both cases it is safe to remove items with unused prefix.

Reply:
Or even better, I think we can have a separate flag for cleaning. It can also be used to solve a problem in the comment above.